### PR TITLE
Ruby 2.4.0 and Minitest 6 support

### PR DESCRIPTION
**Pull request by iMacTia**
WIP to get Faraday supporting both Ruby 2.4 (currently RC1 available) and Minitest 6.
Fixes some deprecation warnings.

To be merged only after Ruby 2.4.0 gets his final release

### CODE REVIEW

**Comment by koic** [comment]
Ruby 2.4.0 has been released 🎉  Thanks.

https://www.ruby-lang.org/en/news/2016/12/25/ruby-2-4-0-released

File: .travis.yml [hunk 1]
```diff
@@ -8,7 +8,8 @@ rvm:
   - 2.0.0
   - 2.1
   - 2.2
-  - 2.3.1
+  - 2.3.0
+  - 2.4.0-rc1
```
Comment (koic) [inline .travis.yml]: Ruby 2.4.0 available on Travis CI.

http://rubies.travis-ci.org

**Comment by morgoth** [comment]
Do you think we can have release with Ruby 2.4 warnings removed?

**Comment by iMacTia** [comment]
Hi @morgoth, this fix was released with version 0.11.0 so warnings should now be removed.
Are you still seeing other warnings? If so, can you please give us some additional details?

**Comment by morgoth** [comment]
No, all fine :-) - I thought I was using latest faraday, but there was version restriction from other gem, so I was still on 0.10.0. Thanks

**Pull request merged**